src/modes/coincident_select.js

```diff
@@ -83,13 +83,30 @@ CoincidentSelect.onSetup = function(opts) {
   const planId = features.find(f => f.properties.vetro_id === feature.id)
     .properties.plan_id;
 
-  features.forEach(f => {
+  for(const f of features){
     if (
       f.properties.plan_id === planId &&
       f.geometry.type === "LineString" &&
       !f.layer.id.includes("_snap")
     ) {
-      const lineGeom = JSON.parse(f.properties.geojson_string);
+
+      let lineGeom = f.geometry;
+      if(typeof this._ctx.options.fetchSourceGeometry === "function"){
+
+        const [ lineSrcGeom, ptSrcGeom ] = await Promise.allSettled([
+          this._ctx.options.fetchSourceGeometry(f),
+          this._ctx.options.fetchSourceGeometry({ properties: { vetro_id: state.initiallySelectedFeatureIds[0] }})
+        ]);
+
+        if(lineSrcGeom && lineSrcGeom.type && lineSrcGeom.coordinates.length){
```

Comment:
are there circumstances under which promises will resolve but these conditions will be false?

Reply:
Since the returned values from the promises are based on whatever the user defines in their `options` `fetch` function, I figured it would be good to make sure that the values are useable.